packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx

```diff
@@ -172,11 +181,9 @@ export class AtomicSearchBox implements AtomicSearchBoxOptions {
           onMouseDown={(e) => e.preventDefault()}
           part="suggestion"
           id={id}
-          value={suggestion.value}
-        >
-          {/* Add highlighting */}
-          {suggestion.value}
-        </button>
+          value={suggestion.rawValue}
+          innerHTML={suggestion.value}
```

Comment:
Hmmm, one thing I just realized from this line ... it's not possible to defend against XSS when the html tags are formatted inside the controller. I _think_ we need to go back to something similar to your initial design exposing two values and requiring the user to pass the correct value 😞 

```
const valuesToHighlight = searchBox.state.formattableValues;
const sanitizedValues = valuesToHighlight.map(v => escape(v));
const highlightedValues = sanitizedValues.map(v => searchBox.highlight({v, options});
```

Let's discuss in the scrum tomorrow. Maybe someone on the team has a better idea,

Reply:
Couldn't we sanitize inside the controller?

Reply:
Ideally, we want headless to not assume that we are dealing with html and just pass the data as it is. If we escape html, this will work in browsers, but may not be enough/incorrect when running inside a mobile app.

Headless is in a similar position as the the search-api which does not sanitize the content it sends in the response for the same reason.

With that said, we'll see what the rest of the team thinks,

Reply:
I spoke with the api team, and they are planning to sanitize text in a future version. Until we switch, let's escape query highlighted query suggestions. I think the best place to put this would be in the highlight utility instead of the controller.

Reply:
I update `highlight.ts` to escape the string first thing in `getHighlightedSuggestion`. There was 2 possible escape function available, but they were escaping to hexadecimal values. Instead, I created a small escape function (like underscore.escape).